### PR TITLE
Initial tests for celery autoscaler.

### PR DESCRIPTION
**Pull request by emord**
@gcapalbo I mentioned yesterday that I would add tests

buddy @esoergel

### CODE REVIEW

File: corehq/util/tests/test_celery_utils.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import mock
+
+from corehq.util.celery_utils import LoadBasedAutoscaler, OffPeakLoadBasedAutoscaler
+
+
+class Object(object):
+    pass
+
+
+class MockPool(BasePool):
```
Comment (emord) [inline corehq/util/tests/test_celery_utils.py]: This setup is pulled from celery tests https://github.com/celery/celery/blob/v3.1.25/celery/tests/worker/test_autoscale.py

**Review by gcapalbo — APPROVED** [review]
thanks!

**Comment by esoergel** [comment]
```
======================================================================
FAIL: corehq.util.tests.test_celery_utils:TestOffPeakLoadBasedScaler.test_fewer_processes_than_tasks_off_peak
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/django/test/utils.py", line 205, in inner
    return func(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/util/tests/test_celery_utils.py", line 112, in test_fewer_processes_than_tasks_off_peak
    self.assert_scale_up(0, 4)
  File "/mnt/commcare-hq-ro/corehq/util/tests/test_celery_utils.py", line 48, in assert_scale_up
    self.assertTrue(x._maybe_scale())
AssertionError: None is not true
----------------------------------------------------------------------
Ran 958 tests in 100.043s
FAILED (SKIP=249, failures=1)
```
and 
```
======================================================================
FAIL: corehq.util.tests.test_celery_utils:TestLoadBasedScaler.test_fewer_processes_than_tasks
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/util/tests/test_celery_utils.py", line 76, in test_fewer_processes_than_tasks
    self.assert_scale_up(0, 4)
  File "/mnt/commcare-hq-ro/corehq/util/tests/test_celery_utils.py", line 48, in assert_scale_up
    self.assertTrue(x._maybe_scale())
AssertionError: None is not true
----------------------------------------------------------------------
Ran 908 tests in 32.093s
FAILED (SKIP=273, failures=1)
Exception TypeError: "'NoneType' object is not iterable" in <bound method SimpleProducer.__del__ of <SimpleProducer batch=False>> ignored
```

**Comment by emord** [comment]
Weird. these are passing locally for me. will have to dig into it more

**Comment by emord** [comment]
This autoscaler didn't work well on prod. Closing this out and will be removing the code in a later PR

**Pull request closed without merging**